[2023-10-11T19:27:43.829+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-11T19:27:43.850+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-11T19:27:43.851+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-11T19:27:43.869+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-03 00:00:00+00:00
[2023-10-11T19:27:43.877+0000] {standard_task_runner.py:57} INFO - Started process 655 to run task
[2023-10-11T19:27:43.884+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-03T00:00:00+00:00', '--job-id', '22', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmphlhhvxeg']
[2023-10-11T19:27:43.889+0000] {standard_task_runner.py:85} INFO - Job 22: Subtask trusted
[2023-10-11T19:27:43.968+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [running]> on host f711e14c143e
[2023-10-11T19:27:44.057+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-03T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-03T00:00:00+00:00'
[2023-10-11T19:28:39.566+0000] {local_task_job_runner.py:294} WARNING - State of this instance has been externally set to None. Terminating instance.
[2023-10-11T19:28:39.569+0000] {process_utils.py:131} INFO - Sending 15 to group 655. PIDs of all processes in the group: [666, 655]
[2023-10-11T19:28:39.569+0000] {process_utils.py:86} INFO - Sending the signal 15 to group 655
[2023-10-11T19:28:39.569+0000] {taskinstance.py:1630} ERROR - Received SIGTERM. Terminating subprocesses.
[2023-10-11T19:28:39.577+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/wikipedia_etl.py", line 133, in trusted
    sc = SparkContext(conf=conf)
  File "/home/airflow/.local/lib/python3.8/site-packages/pyspark/context.py", line 201, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.8/site-packages/pyspark/context.py", line 436, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
  File "/home/airflow/.local/lib/python3.8/site-packages/pyspark/java_gateway.py", line 104, in launch_gateway
    time.sleep(0.1)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1632, in signal_handler
    raise AirflowException("Task received SIGTERM signal")
airflow.exceptions.AirflowException: Task received SIGTERM signal
[2023-10-11T19:28:39.580+0000] {taskinstance.py:1398} INFO - Marking task as UP_FOR_RETRY. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231003T000000, start_date=20231011T192743, end_date=20231011T192839
[2023-10-11T19:28:39.590+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 22 for task trusted ((psycopg2.errors.ForeignKeyViolation) insert or update on table "task_fail" violates foreign key constraint "task_fail_ti_fkey"
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(wikipedia_etl, trusted, scheduled__2023-10-03T00:00:00+00:00, -1) is not present in table "task_instance".

[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s) RETURNING task_fail.id]
[parameters: {'task_id': 'trusted', 'dag_id': 'wikipedia_etl', 'run_id': 'scheduled__2023-10-03T00:00:00+00:00', 'map_index': -1, 'start_date': datetime.datetime(2023, 10, 11, 19, 27, 43, 829771, tzinfo=Timezone('UTC')), 'end_date': datetime.datetime(2023, 10, 11, 19, 28, 39, 579859, tzinfo=Timezone('UTC')), 'duration': 55}]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 655)
[2023-10-11T19:28:39.622+0000] {process_utils.py:79} INFO - Process psutil.Process(pid=655, status='terminated', exitcode=1, started='19:27:43') (655) terminated with exit code 1
[2023-10-11T19:28:39.622+0000] {process_utils.py:79} INFO - Process psutil.Process(pid=666, status='terminated', started='19:27:43') (666) terminated with exit code None
[2023-10-11T19:32:23.102+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-11T19:32:23.109+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-11T19:32:23.109+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-11T19:32:23.120+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-03 00:00:00+00:00
[2023-10-11T19:32:23.124+0000] {standard_task_runner.py:57} INFO - Started process 1464 to run task
[2023-10-11T19:32:23.127+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-03T00:00:00+00:00', '--job-id', '28', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmpayvv30i_']
[2023-10-11T19:32:23.129+0000] {standard_task_runner.py:85} INFO - Job 28: Subtask trusted
[2023-10-11T19:32:23.164+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [running]> on host f711e14c143e
[2023-10-11T19:32:23.221+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-03T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-03T00:00:00+00:00'
[2023-10-11T19:32:38.810+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-11T19:32:38.819+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231003T000000, start_date=20231011T193223, end_date=20231011T193238
[2023-10-11T19:32:38.844+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-11T19:32:38.857+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-11T19:47:25.159+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-11T19:47:25.166+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-11T19:47:25.167+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-11T19:47:25.177+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-03 00:00:00+00:00
[2023-10-11T19:47:25.182+0000] {standard_task_runner.py:57} INFO - Started process 4104 to run task
[2023-10-11T19:47:25.184+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-03T00:00:00+00:00', '--job-id', '48', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmp6y5ubyy_']
[2023-10-11T19:47:25.187+0000] {standard_task_runner.py:85} INFO - Job 48: Subtask trusted
[2023-10-11T19:47:25.224+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [running]> on host f711e14c143e
[2023-10-11T19:47:25.286+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-03T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-03T00:00:00+00:00'
[2023-10-11T19:47:54.817+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-11T19:47:54.828+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231003T000000, start_date=20231011T194725, end_date=20231011T194754
[2023-10-11T19:47:54.868+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-11T19:47:54.879+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-13T17:18:11.971+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-13T17:18:11.985+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-13T17:18:11.986+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-13T17:18:12.009+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-03 00:00:00+00:00
[2023-10-13T17:18:12.019+0000] {standard_task_runner.py:57} INFO - Started process 610 to run task
[2023-10-13T17:18:12.024+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-03T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmp6zx7d9x1']
[2023-10-13T17:18:12.028+0000] {standard_task_runner.py:85} INFO - Job 8: Subtask trusted
[2023-10-13T17:18:12.100+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [running]> on host feba906f3341
[2023-10-13T17:18:12.236+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-03T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-03T00:00:00+00:00'
[2023-10-13T17:18:58.495+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-13T17:18:58.519+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231003T000000, start_date=20231013T171811, end_date=20231013T171858
[2023-10-13T17:18:58.569+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-13T17:18:58.604+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-16T11:25:54.110+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-16T11:25:54.127+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-16T11:25:54.127+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-16T11:25:54.146+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-03 00:00:00+00:00
[2023-10-16T11:25:54.155+0000] {standard_task_runner.py:57} INFO - Started process 584 to run task
[2023-10-16T11:25:54.159+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-03T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmp14rqpgyz']
[2023-10-16T11:25:54.163+0000] {standard_task_runner.py:85} INFO - Job 8: Subtask trusted
[2023-10-16T11:25:54.220+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [running]> on host ab44499fca46
[2023-10-16T11:25:54.314+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-03T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-03T00:00:00+00:00'
[2023-10-16T11:26:33.714+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-16T11:26:33.725+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231003T000000, start_date=20231016T112554, end_date=20231016T112633
[2023-10-16T11:26:33.751+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-16T11:26:33.768+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2023-10-16T19:09:06.172+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-16T19:09:06.183+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [queued]>
[2023-10-16T19:09:06.183+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 3
[2023-10-16T19:09:06.203+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): trusted> on 2023-10-03 00:00:00+00:00
[2023-10-16T19:09:06.210+0000] {standard_task_runner.py:57} INFO - Started process 577 to run task
[2023-10-16T19:09:06.214+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'wikipedia_etl', 'trusted', 'scheduled__2023-10-03T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/wikipedia_etl.py', '--cfg-path', '/tmp/tmpbzoqkrti']
[2023-10-16T19:09:06.217+0000] {standard_task_runner.py:85} INFO - Job 8: Subtask trusted
[2023-10-16T19:09:06.260+0000] {task_command.py:415} INFO - Running <TaskInstance: wikipedia_etl.trusted scheduled__2023-10-03T00:00:00+00:00 [running]> on host 3522330161cc
[2023-10-16T19:09:06.327+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='wikipedia_etl' AIRFLOW_CTX_TASK_ID='trusted' AIRFLOW_CTX_EXECUTION_DATE='2023-10-03T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-03T00:00:00+00:00'
[2023-10-16T19:09:45.172+0000] {python.py:194} INFO - Done. Returned value was: None
[2023-10-16T19:09:45.181+0000] {taskinstance.py:1398} INFO - Marking task as SUCCESS. dag_id=wikipedia_etl, task_id=trusted, execution_date=20231003T000000, start_date=20231016T190906, end_date=20231016T190945
[2023-10-16T19:09:45.230+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 0
[2023-10-16T19:09:45.241+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
